File: genetic-packer/src/main/java/genetic/packer/fx/calculation/BoundingSizeCalculator.java
package genetic.packer.fx.calculation;

import java.util.function.Function;

import genetic.packer.fx.specification.BoundsGetter;
import javafx.geometry.Bounds;
import javaslang.Tuple2;
import javaslang.collection.Traversable;

/**
 * @author piotr.larysz
 */
public interface BoundingSizeCalculator extends Function<Tuple2<BoundsGetter, BoundsGetter>, Double> {

    interface Creator {

        BoundingSizeCalculator from(Traversable<Bounds> bounds);
    }
}

File: genetic-packer/src/main/java/genetic/packer/fx/calculation/BoundingSizeCalculator.kt
package genetic.packer.fx.calculation

import genetic.packer.fx.specification.BoundsGetter
import javafx.geometry.Bounds

interface BoundingSizeCalculator : (Pair<BoundsGetter, BoundsGetter>) -> Double {

  interface Creator {

    fun from(bounds: Sequence<Bounds>): BoundingSizeCalculator
  }
}